AK/ObservableProperty.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <AK/Function.h>
+#include <AK/NonnullRefPtr.h>
+#include <AK/Vector.h>
+#include <LibCore/Object.h>
```

Comment:
AK should and can not depend on LibCore. If the observer interacts with the Core::Object system, it doesn't belong in AK.